File: rust_book/05_struct_data/src/triangle.rs
struct Rectangle {
    height: usize,
    width: usize,
}

impl Rectangle {
    // Static Methods, yey
    fn new(height: usize, width: usize) -> Self {
        Self { height, width }
    }

    fn area(&self) -> usize {
        self.width * self.height
    }
}

fn main() {
    let rect = Rectangle::new(50, 60);

    println!(
        "The area of the rectangle is {} square pixels.",
        rect.area()
    );

    let rect_two = &Rectangle { height: 23, ..rect };

    println!(
        "The area of the second rectangle is {} square pixels.",
        rect_two.area()
    );
}

File: book/05_struct_data/src/triangle.c
#include <stdio.h>

struct Rectangle {
  int width;
  int height;
};

int rect_area(struct Rectangle r) { return r.width * r.height; }

int main() {
  struct Rectangle rect = {50, 60};

  printf("The area of the rectangle is %d square pixels.", rect_area(rect));

  return 0;
}
